modules/openshift-developer-cli-sample-applications.adoc

```diff
@@ -0,0 +1,121 @@
+
+----
+$ odo create httpd --git https://github.com/openshift/httpd-ex.git
+----
+
+=== java
+
+This example helps build and run fat JAR Java applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/fabric8io-images/s2i/blob/master/README.md[Java S2I Builder image].
+
+----
+$ odo create java --git https://github.com/spring-projects/spring-petclinic.git
+----
+
+=== nodejs
+
+Build and run Node.js applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/sclorg/s2i-nodejs-container/blob/master/8/README.md[Node.js 8 container image].
+
+----
+$ odo create nodejs --git https://github.com/openshift/nodejs-ex.git
+----
+
+=== perl
+
+This example helps build and run Perl applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/sclorg/s2i-perl-container/blob/master/5.26/README.md[Perl 5.26 container image].
+
+----
+$ odo create perl --git https://github.com/openshift/dancer-ex.git
+----
+
+=== php
+
+This example helps build and run PHP applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/sclorg/s2i-php-container/blob/master/7.1/README.md[PHP 7.1 Docker image].
+
+----
+$ odo create php --git https://github.com/openshift/cakephp-ex.git
+----
+
+=== python
+
+This example helps build and run Python applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/sclorg/s2i-python-container/blob/master/3.6/README.md[Python 3.6 container image].
+
+----
+$ odo create python --git https://github.com/openshift/django-ex.git
+----
+
+=== ruby
+
+This example helps build and run Ruby applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see link:https://github.com/sclorg/s2i-ruby-container/blob/master/2.5/README.md[Ruby 2.5 container image].
+
+----
+$ odo create ruby --git https://github.com/openshift/ruby-ex.git
+----
+
+=== wildfly
+
+This example helps build and run WildFly applications on CentOS 7. For more information about using this builder image, including {product-title} considerations, see the link:https://github.com/wildfly/wildfly-s2i/blob/master/README.md[Wildfly - CentOS Docker images for OpenShift].
+
+----
+$ odo create wildfly --git https://github.com/openshift/openshift-jee-sample.git
+----
+
+== Binary examples
+
+=== java
+
+Java can be used to deploy a binary artifact, for example:
+
+----
+$ git clone https://github.com/spring-projects/spring-petclinic.git
+$ cd spring-petclinic
+$ mvn package
```

Comment:
It breaks in the middle. I have already created the issue (https://github.com/spring-projects/spring-petclinic/issues/458) in the upstream repo.

Reply:
It works on my setup. Probably something went wrong in your setup while doing ```mvn package```

Reply:
@prietyc123 it works with my setup too (on two computers as well...)

Reply:
In your command you put `$ ./mvnw package` ?

Reply:
If that requires documenting, add it to the KIs.

Reply:
I got the response from the upstream issue that Java 13 is not supported by Spring Boot 2.1.x and java 11 should fix the issue as per the upstream response. @cdrage @amitkrout It seems you have the supported version on your host for the application ```spring-petclinic``` .

See https://github.com/spring-projects/spring-petclinic/issues/458#issuecomment-540077784 for the detailed response from upstream side.

IMO if this is the case then we should mention the supported java version for this application. WDYT?

Reply:
We have already an existing issue on repo spring-petclinic where it fails on the latest java version. However in the same issue its already been addressed how to handle it.

See  https://github.com/spring-projects/spring-petclinic/issues/458#issuecomment-540077784 . So mentioning the same thing over here doesn't make more sense to me.